app/Commands/EditEnvironmentVariables.php

```diff
@@ -0,0 +1,174 @@
+    }
+
+    private function getMenuItem(string $bucket): ?int
+    {
+        $this->envFiles = collect($this->helpers->aws()->s3()->listFiles($this, $bucket))
+            ->pluck('Key')
+            ->map(function ($filename) {
+                return $filename;
+            })
+            ->reject(function ($filename) {
+                return !Str::endsWith($filename, '.env');
+            })
+            ->toArray();
+
+        $additionalMenuOptions = ['Create new file'];
+        $callerArn = $this->helpers->aws()->iam()->getCallerArn($this);
+        $callerArnParts = explode('/', $callerArn);
+
+        if ($callerArnParts[1] == 'NetsellsSecurityOps') {
+            $additionalMenuOptions[] = 'Delete file';
+        }
+
+        return $this->menu("Choose a file to edit", array_merge($this->envFiles, $additionalMenuOptions))->open();
+    }
+
+    private function getFile(int $menuItem, string $bucket): void
+    {
+        if ($menuItem < count($this->envFiles)) {
+            $this->fileName = $this->envFiles[$menuItem];
+            $this->fileContents = (string)$this->helpers->aws()->s3()->getFile($this, $bucket, $this->fileName)->get('Body');
+        } elseif ($menuItem == count($this->envFiles)) {
+            $this->fileName = $this->ask('Enter name of new file (must have an extension of .env)');
+            if (!Str::endsWith($this->fileName, '.env')) {
+                exit(1);
+            }
+            $this->fileContents = '';
+        } else {
+            $deleteMenuItem = $this->menu("Choose a file to edit",$this->envFiles)->open();
+            if ($deleteMenuItem !== null) {
+                if ($this->confirm("Are you sure you wish to delete {$this->envFiles[$deleteMenuItem]}?")) {
+                    $this->helpers->aws()->s3()->deleteFile($this, $bucket, $this->envFiles[$deleteMenuItem]);
+                }
+            }
+            exit(0);
+        }
+
+        file_put_contents('/tmp/' . $this->fileName, $this->fileContents);
+    }
+
+    private function editFile(): string
+    {
+        switch ($pid = pcntl_fork()) {
+            case -1:
+                die('Unable to run editor');
+            case 0:
+                pcntl_exec(config('app.editor'), ['/tmp/' . $this->fileName]);
+                break;
+            default:
+                pcntl_waitpid($pid, $status);
+                break;
+        }
+
+        return file_get_contents('/tmp/' . $this->fileName);
+    }
+
+    private function checkDiffAndUpload(string $newContents, string $bucket): void
+    {
+        $diff = xdiff_string_diff($this->fileContents, $newContents);
+
+        if ($diff == '') {
+            $this->info('No changes made.');
+            exit(0);
+        }
+
+        $this->info("Changes made\n\n");
+
+        echo($diff);
+
+        if ($this->confirm('Do you want to upload the changes? (Y/N)')) {
+            $this->helpers->aws()->s3()->putFile($this, $bucket, $this->fileName, '/tmp/' . $this->fileName);
```

Comment:
A confirmation `->info()` to let the user know this was a success here would be good

Reply:
done